backend/controller/observability/async_calls.go

```diff
@@ -84,19 +97,28 @@ func (m *AsyncCallMetrics) Executed(ctx context.Context, verb schema.RefKey, ori
 }
 
 func (m *AsyncCallMetrics) Completed(ctx context.Context, verb schema.RefKey, origin string, scheduledAt time.Time, maybeErr error) {
-	m.completed.Add(ctx, 1, metric.WithAttributes(extractAsyncCallAndMaybeErrAttrs(verb, origin, scheduledAt, maybeErr)...))
-}
+	msToComplete := timeSince(scheduledAt)
 
-func extractAsyncCallAndMaybeErrAttrs(verb schema.RefKey, origin string, scheduledAt time.Time, maybeErr error) []attribute.KeyValue {
-	attrs := extractAsyncCallAttrs(verb, origin, scheduledAt)
-	return append(attrs, attribute.Bool(observability.StatusSucceededAttribute, maybeErr == nil))
+	attrs := extractRefAttrs(verb, origin)
+	attrs = append(attrs, attribute.Bool(observability.StatusSucceededAttribute, maybeErr == nil))
+	m.msToComplete.Record(ctx, msToComplete, metric.WithAttributes(attrs...))
+
+	attrs = append(attrs, attribute.Int64(asyncCallTimeSinceScheduledAtAttr, msToComplete))
+	m.completed.Add(ctx, 1, metric.WithAttributes(attrs...))
 }
 
 func extractAsyncCallAttrs(verb schema.RefKey, origin string, scheduledAt time.Time) []attribute.KeyValue {
+	return append(extractRefAttrs(verb, origin), attribute.Int64(asyncCallTimeSinceScheduledAtAttr, timeSince(scheduledAt)))
+}
+
+func extractRefAttrs(verb schema.RefKey, origin string) []attribute.KeyValue {
 	return []attribute.KeyValue{
 		attribute.String(observability.ModuleNameAttribute, verb.Module),
 		attribute.String(asyncCallVerbRefAttr, verb.String()),
 		attribute.String(asyncCallOriginAttr, origin),
-		attribute.Int64(asyncCallTimeSinceScheduledAtAttr, time.Since(scheduledAt).Milliseconds()),
 	}
 }
+
+func timeSince(start time.Time) int64 {
+	return time.Since(start).Milliseconds()
+}
```

Comment:
nit: Do we need this func? Seems like it's only used in the one place above which could just call `time.Since(start).Milliseconds()` directly.

Reply:
It's used twice - `extractAsyncCallAttrs` and `Completed`. I was waffling over whether this was worth including as I was writing it though... what do you think?

Reply:
Oh sorry, missed that one! :)

I think if we want to keep the units here consistent across all metrics, maybe we should add this to `observability.go` to be shared with all metrics?

```go
func millisecondsSince(start time.Time) int64 {
	return time.Since(start).Milliseconds()
}
```

It looks like it might be used in a few other places for metrics as well. This can also happen in another PR if you prefer.

Reply:
Good call, I'll pull it into observability.go and leave the refactor of the existing time code for another PR